refactor(linter/plugins): remove `require(esm)`

overlookmotel opened the pull request:
`loadPlugin` is already async, so we don't need to use `require(esm)` for lazy-loading the JS plugins-related code. Use `import` instead.

This has 3 advantages:

1. Removes the hacky workaround of a separate entry point for TSDown, and hard-coded path in `require('./plugins.js')`, which is only correct if TSDown config remains unchanged.
2. TypeScript can "see into" the `import` -> better type safety.
3. Support versions of NodeJS which don't support `require(esm)`.

On the last one, it's not strictly necessary, as `oxlint` requires a version of NodeJS with `require(esm)` support:

https://github.com/oxc-project/oxc/blob/b622ef87adf1bbed1e76ba0f610b653acb93e72b/apps/oxlint/package.json#L19-L21

But if user *is* on an older version of NodeJS, JS plugins should now work.

overlookmotel commented on apps/oxlint/src-js/cli.ts:
```diff
@@ -15,9 +17,12 @@ let lintFile: typeof lintFileWrapper | null = null;
+      ({ loadPlugin, lintFile } = mod);
+      return loadPlugin(path, packageName);
+    });
   }
```
`loadPluginWrapper` is only called once from Rust side, and its `Promise` is awaited before continuing. So this race can't happen. In any case, even if it *did* happen, NodeJS's module loader would deal with it gracefully.

camc314 commented:
### Merge activity

* **Nov 19, 11:08 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 19, 11:08 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
refactor(linter/plugins): remove `require(esm)` (#15866)

`loadPlugin` is already async, so we don't need to use `require(esm)` for lazy-loading the JS plugins-related code. Use `import` instead.

This has 3 advantages:

1. Removes the hacky workaround of a separate entry point for TSDown, and hard-coded path in `require('./plugins.js')`, which is only correct if TSDown config remains unchanged.
2. TypeScript can "see into" the `import` -> better type safety.
3. Support versions of NodeJS which don't support `require(esm)`.

On the last one, it's not strictly necessary, as `oxlint` requires a version of NodeJS with `require(esm)` support:

https://github.com/oxc-project/oxc/blob/b622ef87adf1bbed1e76ba0f610b653acb93e72b/apps/oxlint/package.json#L19-L21

But if user *is* on an older version of NodeJS, JS plugins should now work.

camc314 commented:
yep this was just a local sanity check to make sure we were going to be including everything that was needed in the published packages
